### PR TITLE
Adds test cases for imports in Ion Schema 2.0

### LINKED ISSUE #4: Add test verifying self-import is not allowed
Any given schema that attempts to import itself should result in an error.  Note that this is a special case of a circular import (A imports A; vs. A imports B, which imports C, which imports A).

_Originally posted by @therapon in https://github.com/amzn/ion-schema-tests/pull/3_

### PR DESCRIPTION
**Pull request by popematt**
**Issue #, if available:**

#32 

Also fixes #4 

**Description of changes:**

Adds test cases for import syntax and for resolving imports.

_By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice._

### CODE REVIEW

File: ion_schema_2_0/imports/invalid_imports.isl [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+        name: foo,
+        type: { id: "util.isl", id: "util.isl", type: positive_int }
+      }
+    ),
+    (
+      $ion_schema_2_0
+      type::{
+        name: foo,
+        type: { id: "util.isl", type: positive_int, type: positive_int }
+      }
```
Comment (tgregg) [inline ion_schema_2_0/imports/invalid_imports.isl:166]: This is duplicated at line 108.

File: ion_schema_2_0/imports/isl_1_0_importing_isl_2_0.isl [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+}
+
+type::{
+  // This is using features not available in Ion Schema 1.0
+  name: snake_case_annotated,
+  annotations: { element: { regex: "^[a-z0-9_]+$"} },
+}
+
+$test::{
+  description: "Ion Schema 1.0 should be able to import from an Ion Schema 2.0 document",
```
Comment (tgregg) [inline ion_schema_2_0/imports/isl_1_0_importing_isl_2_0.isl:15]: What actually happens? Should we add some assertions?

File: ion_schema_2_0/imports/isl_2_0_importing_isl_1_0.isl [hunk 3]
```diff
@@ -0,0 +1,25 @@
+$ion_schema_1_0
+
+type::{
+  name: ordinary_struct,
+  type: struct,
+}
+
+$test::{
+  description: "Ion Schema 2.0 should be able to import from an Ion Schema 1.0 document",
```
Comment (tgregg) [inline ion_schema_2_0/imports/isl_2_0_importing_isl_1_0.isl:9]: Should we add some assertions?

File: ion_schema_2_0/imports/tree/header_import_a.isl [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  annotations: { valid_values: [[a]] },
+  fields: closed::{
+    b: b_type,
+    c: c_type,
+  },
+}
+
+$test::{
+  type: a_type,
+  should_accept_as_valid: [
```
Comment (tgregg) [inline ion_schema_2_0/imports/tree/header_import_a.isl:21]: Consider adding one or two values that should be rejected.

**Review by tgregg — APPROVED** [review]

File: ion_schema_2_0/imports/header_imports.isl [hunk 5]
```diff
@@ -0,0 +1,296 @@
...
+        name: positive_int,
+        type: positive_int_1
+      }
+      schema_footer::{}
+    ),
+  ]
+}
+
+$test::{
+  description: "A schema should not be able to reference an aliased type by it's original name",
```
[the remark this answers is not in the document]
Comment (popematt) [reply ion_schema_2_0/imports/header_imports.isl:141]:
```suggestion
  description: "A schema should not be able to reference an aliased type by its original name",
```

File: ion_schema_2_0/imports/isl_1_0_importing_isl_2_0.isl (hunk 2 above)
Comment (popematt) [reply ion_schema_2_0/imports/isl_1_0_importing_isl_2_0.isl:15]: Sure. I'm going to refactor these tests so they can assert that the types do what we expect them to do.

File: ion_schema_2_0/imports/isl_2_0_importing_isl_1_0.isl (hunk 3 above)
Comment (popematt) [reply ion_schema_2_0/imports/isl_2_0_importing_isl_1_0.isl:9]: Will do.

File: ion_schema_2_0/imports/diamond/header_import_a.isl [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+  ]
+}
+
+schema_footer::{}
+
+$test::{
+  description: "no indirectly imported types should be available in this schema's scope",
+  invalid_types: [
+    { type: d_type, },
+  ]
```
[the remark this answers is not in the document]
Comment (popematt) [reply ion_schema_2_0/imports/diamond/header_import_a.isl:46]: I've added some test cases for that now.

**Review by tgregg — APPROVED** [review]

**Pull request merged**